calling-apis/get-requests-with-axios.md

```diff
@@ -0,0 +1,453 @@
+      error.response.status
+    );
+
+    console.log(
+      "The data given back by the API response is:",
+      error.response.data
+    );
+  });
+```
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: b27e89d8
+* title: GET Requests With axios
+##### !question
+
+Which of the following options will give back the HTTP response body, which is the data that Jamel is expecting from the API?
+
+##### !end-question
+##### !options
+
+* `response`
+* `response.status`
+* `response.headers.date`
+* `response.data`
+
+##### !end-options
+##### !answer
+
+* `response.data`
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### Verifying the Error Handling
+
+How can we check that an unsuccessful response goes into the `catch` block? One approach we can take is to send a request to an invalid URL.
+
+Let's change the URL to `'https://dog.ceo/api/breeds/notarealendpoint/'`, so we can look at what happens when we get an error response.
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: 7cbb2b75
+* title: GET Requests With axios
+##### !question
+
+In the case of handling an unsuccessful request, which of the following options gives back the HTTP response body, which contains the data that the API sent back?
+
+##### !end-question
+##### !options
+
+* `error`
+* `error.response.status`
+* `error.response.data`
+
+##### !end-options
+##### !answer
+
+* `error.response.data`
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### `finally`
+
+`axios` supports a `finally` clause, which will run after either the `then` or `catch` no matter what.
+
+The example syntax looks like this:
+
+```js
+const axios = require("axios");
+
+axios
+  .get("https://dog.ceo/api/breeds/list/all")
+  .then((response) => {
+    console.log("success!");
+  })
+  .catch((error) => {
+    console.log("error!");
+  })
+  .finally(() => {
+    console.log("this is always executed, no matter what!");
+  });
+```
+
+## Query Params
+
+As part of Jamel's DogSpotter app, he needs to find the latitude and longitude of Seattle, Washington, USA. He'll use the [LocationIQ Geocoding API](https://locationiq.com/sandbox/geocoding/forward).
+
+After looking at [the docs for Search/Forward Geocoding](https://locationiq.com/docs), Jamel determined that his request should look like this:
+
+- `GET` request to `https://us1.locationiq.com/v1/search.php`
+- Query Params:
+  - `key`: the API key
+  - `q`: A search term. In this case, `"Seattle, Washington, USA"`
+  - `format`: `"json"`
+
+He expects a response body that looks like:
+
+```json
+[
+  {
+    "place_id": "332439918643",
+    "licence": "https://locationiq.com/attribution",
+    "lat": "47.673455",
+    "lon": "-122.331537",
+    "display_name": "Seattle, King County, Washington, USA",
+    "boundingbox": ["47.495514", "47.734165", "-122.435956", "-122.236044"],
+    "importance": 0.25
+  }
```

Comment:
looks like this might be an old request? not sure what caused some of the lat/lon changes (my guess is that maybe they're using a different reference ellipsoid???)

Feel free to cut down some of the fields in this response if they are more noisy than helpful.

Also, I noticed that even when making the exact same request repeated times, I would sometimes gets 1 result, and other times several (some stuff in Kent and elsewhere).


```suggestion
  {
    "place_id": "235549103",
    "licence": "https://locationiq.com/attribution",
    "osm_type": "relation",
    "osm_id": "237385",
    "boundingbox": [ "47.4810022", "47.7341357", "-122.459696", "-122.224433" ],
    "lat": "47.6038321",
    "lon": "-122.3300624",
    "display_name": "Seattle, King County, Washington, USA",
    "class": "place",
    "type": "city",
    "importance": 0.8929791735643791,
    "icon": "https://locationiq.org/static/images/mapicons/poi_place_city.p.20.png"
  },

  // possibly other matches
```